src/Routes/Filter/RoutesFilter.php

```diff
@@ -0,0 +1,189 @@
+            return $this->matchesByMiddleware($route);
+        }
+
+        return false;
+    }
+
+    /**
+     * Match the route against a regular expression based on the field in $matchType.
+     *
+     * $matchType can be either "path" or "name".
+     */
+    public function matchesByRegex(Route $route, string $routeAttribute): bool
+    {
+        if ($routeAttribute === 'path') {
+            $path = (string) $this->filterOptions['path'];
+            return (bool) preg_match(
+                sprintf("/^%s/", str_replace('/', '\/', $path)),
+                $route->getPath()
+            );
+        }
+
+        if ($routeAttribute === 'name') {
+            return (bool) preg_match(
+                sprintf(
+                    "/%s/",
+                    (string) $this->filterOptions['name']
+                ),
+                $route->getName()
+            );
+        }
+
+        return false;
+    }
+
+    /**
+     * Match if the current route supports the method(s) supplied.
+     */
+    public function matchesByMethod(Route $route): bool
+    {
+        if ($route->allowsAnyMethod()) {
+            return true;
+        }
+
+        if ($this->filterOptions['method'] === Route::HTTP_METHOD_ANY) {
+            return true;
+        }
+
+        if (is_string($this->filterOptions['method'])) {
+            return in_array(
+                strtoupper($this->filterOptions['method']),
+                $route->getAllowedMethods() ?? []
+            );
+        }
+
+        if (is_array($this->filterOptions['method'])) {
+            array_walk(
+                $this->filterOptions['method'],
+                fn(string &$value) => $value = strtoupper($value)
+            );
+            return ! empty(array_intersect(
+                $this->filterOptions['method'],
+                $route->getAllowedMethods() ?? []
+            ));
+        }
+
+        return false;
+    }
+
+    /**
+     * This method checks if a route is handled by a given middleware class
+     *
+     * The function first checks if there is an exact match on the middleware
+     * class' name, then a partial match to any part of the class' name, and
+     * finally uses a regular expression to attempt a pattern match against
+     * the class' name. The intent is to perform checks from the least to the
+     * most computationally expensive, to avoid excessive overhead.
+     */
+    public function matchesByMiddleware(Route $route): bool
+    {
+        $middlewareClass   = get_class($route->getMiddleware());
+        $matchesMiddleware = (string) $this->filterOptions['middleware'];
+
+        try {
+            return $middlewareClass === $matchesMiddleware
+                || (bool) stripos($middlewareClass, $matchesMiddleware)
+                || (bool) preg_match(
+                    sprintf('/%s/', $matchesMiddleware),
+                    $middlewareClass
+                );
+        } catch (Exception $e) {
```

Comment:
Should probably be `Throwable`? and I think you can drop the `$e` from PHP 8.1 onwards